src/Upgrader/Business/DataProvider/Client/ReleaseApp/ReleaseAppClient.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Upgrader\Business\DataProvider\Client\ReleaseApp;
+
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\HttpCommunicatorInterface;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Request\UpgradeAnalysisRequest;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Request\UpgradeInstructionsRequest;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeAnalysis\Collection\UpgradeAnalysisModuleVersionCollection;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeInstructions\Collection\UpgradeInstructionsReleaseGroupCollection;
+use Upgrader\Business\DataProvider\Entity\Collection\ModuleCollection;
+use Upgrader\Business\DataProvider\Entity\Collection\ReleaseGroupCollection;
+use Upgrader\Business\DataProvider\Entity\Module;
+use Upgrader\Business\DataProvider\Entity\ReleaseGroup;
+use Upgrader\Business\DataProvider\Request\DataProviderRequestInterface;
+use Upgrader\Business\DataProvider\Response\DataProviderResponse;
+
+class ReleaseAppClient implements ReleaseAppClientInterface
+{
+    /**
+     * @var \Upgrader\Business\DataProvider\Client\ReleaseApp\Http\HttpCommunicatorInterface
+     */
+    protected $httpCommunicator;
+
+    /**
+     * @param \Upgrader\Business\DataProvider\Client\ReleaseApp\Http\HttpCommunicatorInterface $httpCommunicator
+     */
+    public function __construct(HttpCommunicatorInterface $httpCommunicator)
+    {
+        $this->httpCommunicator = $httpCommunicator;
+    }
+
+    /**
+     * @param \Upgrader\Business\DataProvider\Request\DataProviderRequestInterface $request
+     *
+     * @return \Upgrader\Business\DataProvider\Response\DataProviderResponse
+     */
+    public function getNotInstalledReleaseGroupList(DataProviderRequestInterface $request): DataProviderResponse
+    {
+        $moduleVersionCollection = $this->getModuleVersionCollection($request);
+        $releaseGroupCollection = $this->getReleaseGroupCollection($moduleVersionCollection);
+        $dataProviderRGCollection = $this->buildDataProviderReleaseGroupCollection($releaseGroupCollection);
+
+        return new DataProviderResponse($dataProviderRGCollection);
+    }
+
+    /**
+     * @param \Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeInstructions\Collection\UpgradeInstructionsReleaseGroupCollection $releaseGroupCollection
+     *
+     * @return \Upgrader\Business\DataProvider\Entity\Collection\ReleaseGroupCollection
+     */
+    protected function buildDataProviderReleaseGroupCollection(
+        UpgradeInstructionsReleaseGroupCollection $releaseGroupCollection
+    ): ReleaseGroupCollection {
+        $dataProviderRGCollection = new ReleaseGroupCollection();
```

Comment:
```suggestion
        $releaseGroupCollection = new ReleaseGroupCollection();
```

Reply:
Thanks!
This PR is not ready for review because we need to merge it with Artems branch 
I notified you when it will be done